models/User.js

```diff
@@ -431,13 +439,32 @@ userSchema.methods.unitLoss = function(lossPercentage) {
 	});
 
 	// Remove hp from your hero depending on the loss percentage
-	this.hero.currentHealth = Math.floor(this.hero.currentHealth * lossPercentage);
+	this.hero.currentHealth = this.hero.currentHealth - Math.floor(this.hero.currentHealth * lossPercentage);
 
+	// if the hero died
+	if (this.hero.currentHealth <= 0 && this.hero.rank > 0) {
+		this.hero.rank -= 1;
+		this.hero.expToNextRank = heroExpToNextLevel[this.hero.rank];
```

Comment:
Issue 2: Når en hero får en ny rank så får den også mer stats. Sjekk funksjonen gainHeroExp i hero-exp.js
Det vil si at hvis vi skal down-ranke hero'en så må vi også fjerne stats, noe som ikke er umulig. Jeg har laget en funksjon for dette her om dagen i henhold til Staking hvor kan mister exp hvis man taper. Sjekk removeHeroExp i hero-exp.js.